### PR TITLE
Enable reporting changes in test mode for cmd states when using stateful output

### PR DESCRIPTION
**Pull request by jond64**
With this commit the stateful argument for cmd states can optionally include a test_name parameter.

This is used to specify a command to run in test mode.  This command should return stateful data for changes that would be made by the command in the name parameter.

``` yaml
keyval1:
  cmd.script:
    - name: nochange.sh Apply
    - source: salt://nochange.sh
    - stateful:
      - test_name: nochange.sh Test

keyval2:
  cmd.script:
    - name: change.sh Apply
    - source: salt://change.sh
    - stateful:
      - test_name: change.sh Test

keyval3:
  cmd.script:
    - name: fail.sh Apply
    - source: salt://fail.sh
    - stateful:
      - test_name: fail.sh Test
```

Note that the Name line in the output is showing the command for the name argument even though I'm setting name in the return dict to the command for the test_name argument when test=True.  Is that preferred for consistency between runs with test=True and test=False?

``` yaml
local:
----------
          ID: keyval1
    Function: cmd.script
        Name: nochange.sh Apply
      Result: True
     Comment: keyval Test
     Started: 14:11:26.081569
    Duration: 10.413 ms
     Changes:
----------
          ID: keyval2
    Function: cmd.script
        Name: change.sh Apply
      Result: None
     Comment: keyval Test
     Started: 14:11:26.092526
    Duration: 8.991 ms
     Changes:
              ----------
              changed:
                  yes
              pid:
                  9881
              retcode:
                  0
              stderr:

              stdout:
                  Test changes ...
----------
          ID: keyval3
    Function: cmd.script
        Name: fail.sh Apply
      Result: False
     Comment: keyval Test
     Started: 14:11:26.101938
    Duration: 9.08 ms
     Changes:
              ----------
              changed:
                  yes
              pid:
                  9882
              retcode:
                  9
              stderr:

              stdout:
                  Test failed ...

Summary
------------
Succeeded: 2 (unchanged=1, changed=2)
Failed:    1
------------
Total states run:     3
```

### CODE REVIEW

File: salt/states/cmd.py [hunk 1]
```diff
@@ -114,6 +114,31 @@
    Note that if the ``cmd.wait`` state also specifies ``stateful: True`` it can
    then be watched by some other states as well.
 
+4. :strong:`The stateful argument can optionally include a test_name parameter.`
+
+   This is used to specify a command to run in test mode.  This command should
+   return stateful data for changes that would be made by the command in the
+   name parameter.
+    
+   .. versionadded:: 2015.3.0
```
Comment (jfindlay) [inline salt/states/cmd.py]: Should be 2015.2.0.  It will be March when the release comes out, but 2015.2.0 is the name we're stuck with. :-)

**Comment by jond64** [comment]
Is this still being reviewed?

**Comment by jfindlay** [comment]
@jond64, I haven't merged this myself as I'm not familiar enough with the code involved, but let's make sure @thatch45 gets back to this one.  Part of the problem was that I forgot to override the test results.  Sometimes one or the other distro test fails because of intermittent problems with the test infrastructure, or even the cloud infrastructure upon which it runs.

**Comment by thatch45** [comment]
Thanks @jond64 ! Sorry this took so long, just got lost in the older PRs. Thanks for pinging us on it!

**Pull request merged**